code/datums/ai/dog/dog_controller.dm

```diff
@@ -0,0 +1,274 @@
+	friends[new_friend] = TRUE
+	RegisterSignal(new_friend, COMSIG_MOB_POINTED, .proc/check_point)
+	RegisterSignal(new_friend, COMSIG_MOB_SAY, .proc/check_command)
+
+/// Someone is being mean to us, take them off our friends (add actual enemies behavior later)
+/datum/ai_controller/dog/proc/unfriend(mob/living/ex_friend)
+	var/list/friends = blackboard[BB_DOG_FRIENDS]
+	friends[ex_friend] = null
+	UnregisterSignal(ex_friend, list(COMSIG_MOB_POINTED, COMSIG_MOB_SAY))
+
+/// Someone we like is pointing at something, see if it's something we might want to interact with (like if they might want us to fetch something for them)
+/datum/ai_controller/dog/proc/check_point(mob/pointing_friend, atom/movable/pointed_movable)
+	SIGNAL_HANDLER
+
+	if(blackboard[BB_FETCH_TARGET] || !ismovable(pointed_movable) || blackboard[BB_DOG_ORDER_MODE] == DOG_COMMAND_NONE)
+		return
+
+	var/list/visible_things = view(pawn, AI_DOG_VISION_RANGE)
+	if(!(pointing_friend in visible_things) || !(pointed_movable in visible_things))
+		return
+
+	switch(blackboard[BB_DOG_ORDER_MODE])
+		if(DOG_COMMAND_FETCH)
+			if(ismob(pointed_movable) || pointed_movable.anchored)
+				return
+			pawn.visible_message("<span class='notice'>[pawn] follows [pointing_friend]'s gesture towards [pointed_movable] and barks excitedly!</span>")
+			current_movement_target = pointed_movable
+			blackboard[BB_FETCH_TARGET] = pointed_movable
+			blackboard[BB_FETCH_DELIVER_TO] = pointing_friend
+			current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/fetch)
+		if(DOG_COMMAND_ATTACK)
+			pawn.visible_message("<span class='notice'>[pawn] follows [pointing_friend]'s gesture towards [pointed_movable] and growls intensely!</span>")
+			current_movement_target = pointed_movable
+			blackboard[BB_DOG_HARASS_TARGET] = pointed_movable
+			current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/harass)
+
+/// Someone is looking at us, if we're currently carrying something then show what it is
+/datum/ai_controller/dog/proc/on_examined(datum/source, mob/user, list/examine_text)
+	SIGNAL_HANDLER
+
+	if(!blackboard[BB_SIMPLE_CARRY_ITEM])
+		return
+
+	var/obj/item/carried_item = blackboard[BB_SIMPLE_CARRY_ITEM]
+	examine_text += "<span class='notice'>[pawn.p_they(TRUE)] [pawn.p_are()] carrying [carried_item.get_examine_string(user)] in [pawn.p_their()] mouth.</span>"
+
+/// If we died, drop anything we were carrying
+/datum/ai_controller/dog/proc/on_death(mob/living/ol_yeller)
+	SIGNAL_HANDLER
+
+	var/obj/item/carried_item = blackboard[BB_SIMPLE_CARRY_ITEM]
+	if(!carried_item)
+		return
+
+	ol_yeller.visible_message("<span='danger'>[ol_yeller] drops [carried_item] as [ol_yeller.p_they()] die[ol_yeller.p_s()].</span>")
+	carried_item.forceMove(get_turf(ol_yeller))
+	blackboard[BB_SIMPLE_CARRY_ITEM] = null
+
+/// One of our friends said something, see if it's a valid command, and if so, take action
+/datum/ai_controller/dog/proc/check_command(mob/speaker, speech_args)
```

Comment:
i'd prefer if this were generalized in some way

Reply:
it's not clear what you mean here, generalised for what? to support more commands, or as part of the AI framework?